modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,80 @@
+// Module included in the following assemblies:
+// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
+// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="ztp-image-based-upgrade-concept_{context}"]
+= Image-based upgrade for {sno} clusters with the {lcao}
+
+You can upgrade the platform version of the managed {sno} clusters with the {lcao}.
+By using generated a seed image, this upgrade process allows you to skip multiple minor releases and it is faster than the standard upgrade method.
+
+You generate the seed image from a cluster that is upgraded and configured with the desired {product-title} and applications by using the standard upgrade method. This is the seed cluster.
+
+[IMPORTANT]
+====
+The image-based upgrade uses custom images that are specific to the hardware platform the clusters are running on.
+Each different hardware platform requires a separate custom image.
+====
+
+With this seed image, you can upgrade the platform version on any {sno} that matches the image's cluster profile and then restore applications.
+
+//How do we specify the hardware exactly? Is it baked into the seed image based on the seed cluster we're using?
+
+The {lcao} uses two custom resources (CRs):
+
+* The `SeedGenerator` CR specifies the platform version of the seed cluster that you want to use to upgrade the target clusters.
+* The `imageBasedUpgrade` CR specifies the generated seed image's location and you use it to step through the upgrade stages.
+
+After generating the seed image, the upgrade process has four stages:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+
+[discrete]
+== Idle stage
+
+There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+[discrete]
+== Prep stage
+
+You can complete this stage in advance.
+
+The {lcao} ensures that the node is ready to proceed to the `Upgrade` stage. This includes the following steps:
+
+* Installing and configuring the OADP Operator
+* Setting up the storage bucket for backup
+* Precaching list of images from the seed image
+
+The {lcao} uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade and rollback.
+
+At this stage, the {lcao} pulls the seed image to the target cluster and two stateroots are created by `ostree`:
+
+* stateroot of the current version of the target cluster
+* stateroot of the newer version that is based on the seed image
+
```

Comment:
Example. See that the default stateroot is called rhcos while the one created by LCA is currently named as rhcos_$ocp_version

~~~sh
[root@snonode ~]# rpm-ostree status -v

State: idle
AutomaticUpdates: disabled
Deployments:  ostree-unverified-registry:quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:4c35b047a35eee940f136034f31c72d08c171da6e4d1ac73dcfee322d9659777 (index: 0)
                   Digest: sha256:4c35b047a35eee940f136034f31c72d08c171da6e4d1ac73dcfee322d9659777
                  Version: 414.92.202312011602-0 (2024-01-15T14:19:36Z)
                   Commit: 7e2137d179196775acf74ffe95150c1fa984b620b77da5c2c211bdcab94b8ab6
                   Staged: no
                StateRoot: rhcos_4.14.6
● ostree-unverified-registry:quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:f4ad0c56ede0a79651cf618601ab0bb93d64a8a15d3d6f82ccb944d6aded7735 (index: 1)
                   Digest: sha256:f4ad0c56ede0a79651cf618601ab0bb93d64a8a15d3d6f82ccb944d6aded7735
                  Version: 414.92.202310270216-0 (2024-01-11T16:12:12Z)
                   Commit: da48dfaea37c2f3380a5d1c1e1b673c68184a9c34581d5fada4833e495dfe04a
                StateRoot: rhcos
~~~

In a future stage (Upgrade IIRC) the default stateroot when booting will be the rhcos_$ocp_version.

Reply:
Too much internals for an end user doc. Admin will initiate the Prep stage by patching the CR and should monitor the status of the CR